api/review_checklist.md

```diff
@@ -0,0 +1,137 @@
+# API Review Checklist
+
+This checklist is intended to be used when reviewing xDS API changes.
+Users who wish to contribute API changes should read this and proactively
+consider the answers to these questions before sending a PR.
```

Comment:
mind linking to this from CONTRIBUTING.md too?

"If your PR includes changes to api/... please read "

Reply:
Done.